From ac7b0aa70ae285252e910fda5a1ef475d7e31dd1 Mon Sep 17 00:00:00 2001
From: Samuel Cabrero <scabrero@suse.de>
Date: Tue, 2 Jul 2019 12:19:00 +0200
Subject: [PATCH] Fix runtests.py

* f-strings require python3.6
* Add slapd to PATH
* kerberos ldap schema path

Signed-off-by: Samuel Cabrero <scabrero@suse.de>
---
 tests/runtests.py | 4 ++--
 tests/testlib.py  | 9 +++++----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/tests/runtests.py b/tests/runtests.py
index 9cd546f..5b7d85e 100755
--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -12,10 +12,10 @@ import testlib
 from testlib import *
 
 def check_exec(name):
-    env = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin'}
+    env = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib/mit/bin:/usr/lib/mit/sbin:/usr/lib/openldap'}
     ret = subprocess.call(["which", name], stdout=subprocess.DEVNULL, env=env)
     if ret != 0:
-        print(f"Executable '{name}' not found in {env['PATH']}",
+        print("Executable '%s' not found in %s" % (name, env['PATH']),
               file=sys.stderr)
         exit(1)
 
diff --git a/tests/testlib.py b/tests/testlib.py
index 05c734d..8d02e5d 100755
--- a/tests/testlib.py
+++ b/tests/testlib.py
@@ -300,8 +300,9 @@ def write_ldap_krb5_config(testdir):
         raise ValueError("Did not find LDAP schemas; is openldap installed?")
 
     k5schema = None
-    for path in ["/usr/share/doc/krb5-server-ldap*/kerberos.schema",
-                 "/usr/share/doc/krb5-kdc-ldap/kerberos.schema.gz"]:
+    # Path in TW and SLE-15-SP1 onward /usr/share/kerberos/ldap
+    for path in ["/usr/share/kerberos/ldap/kerberos.schema",
+                 "/usr/share/doc/packages/krb5/kerberos.schema"]:
         pathlist = glob.glob(path)
         if len(pathlist) > 0:
             k5schema = pathlist[0]
@@ -367,7 +368,7 @@ def setup_ldap(testdir, wrapenv):
     stashfile = os.path.join(testdir, "ldap_passwd")
     krb5conf = os.path.join(testdir, 'krb5.conf')
 
-    ldapenv = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin',
+    ldapenv = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib/openldap:/usr/lib/mit/sbin',
                'KRB5_CONFIG': krb5conf}
     ldapenv.update(wrapenv)
 
@@ -411,7 +412,7 @@ def setup_kdc(testdir, wrapenv):
     kdcstash = os.path.join(kdcdir, KDC_STASH)
     kdcdb = os.path.join(kdcdir, KDC_DBNAME)
 
-    kdcenv = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin',
+    kdcenv = {'PATH': '/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib/mit/sbin:/usr/lib/mit/bin',
               'KRB5_CONFIG': krb5conf,
               'KRB5_KDC_PROFILE': kdcconf}
     kdcenv.update(wrapenv)
-- 
2.22.0

